### PR TITLE
allow for other providers/models and use groq+llama3-70 as default + …

### PR DESCRIPTION
**Pull request by sramanan**
…minor guardrails optimization

### CODE REVIEW

File: groq_script.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from dotenv import load_dotenv
```
[the remark this answers is not in the document]
Comment (sramanan) [reply groq_script.py:1]: dont we? how are the other secrets loaded?

File: gpt_script.py [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+    "openai" : OpenAI(),
+    "groq": Groq(api_key=os.environ.get("GROQ_API_KEY"))
+}
 
 class ChatApp:
     def __init__(
             self,
             conv_limit=30,
             model="gpt-4-turbo",
+            client="openai",
```
[the remark this answers is not in the document]
Comment (sramanan) [reply gpt_script.py:18]: about the same, i'd leave it as it for now

File: gpt_script.py [hunk 3]
```diff
@@ -1,23 +1,30 @@
 from dotenv import load_dotenv
```
[the remark this answers is not in the document]
Comment (sramanan) [reply gpt_script.py:1]: ok

File: groq_script.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vishalsubbiah) [reply groq_script.py:1]: in chat_script it gets imported.

File: gpt_script.py [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
 load_dotenv()  # take environment variables from .env.
 from openai import OpenAI
+from groq import Groq
 from collections import deque
+import os
+
+clients = {
+    "openai" : OpenAI(),
+    "groq": Groq(api_key=os.environ.get("GROQ_API_KEY"))
```
[the remark this answers is not in the document]
Comment (vishalsubbiah) [reply gpt_script.py]: readme changes didn't get pushed?

File: groq_script.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sramanan) [reply groq_script.py:1]: ah, i had this script just as a test for groq. So if i run this standalone i would still need it.

File: gpt_script.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sramanan) [reply gpt_script.py]: yeah, got missed. updating

**Review by vishalsubbiah — APPROVED** [review]

**Comment by nikolaspapastavrou** [comment]
Should we merge this?

**Pull request merged**